src/v/datalake/translation/deps.cc

```diff
@@ -607,40 +585,156 @@ class partition_lag_tracker : public translation_lag_tracker {
+        // cached timestamp, since it provides a lower bound for when
+        // new_offset became translatable, from the perspective of the
+        // translator.
+        // If translation met or exceeded the target, also reset it to nullopt
+        // so that a new target can be set the next time fresh offsets become
+        // available for translation.
+        vlog(
+          datalake_log.trace,
+          "[{}] New offset translated {}, current "
+          "translation_target: {}",
+          _partition->ntp(),
+          translated_offset,
+          _translation_target);
+        if (translated_offset >= _translation_target->offset) {
+            _translation_target.reset();
         }
-        auto target_lag = topic_cfg->properties.iceberg_target_lag_ms.value_or(
-          default_lag);
-        return std::chrono::duration_cast<scheduling::clock::duration>(
-          target_lag);
     }
 
-    void reset_to_next_window() { _current_window_end += target_lag(); }
+    std::optional<size_t> translation_backlog() const final {
+        auto highest_translated_offset
+          = _stm->cached_highest_translated_offset();
+        if (
+          !highest_translated_offset
+          || highest_translated_offset < kafka::offset{0}) {
+            return std::nullopt;
+        }
+        auto min_offset_for_translation = calculate_min_offset_for_translation(
+          _partition->is_read_replica_mode_enabled(), *_partition_proxy);
+        if (highest_translated_offset <= min_offset_for_translation) {
+            return _partition->size_bytes();
+        }
+        auto log_offset = _partition->log()->to_log_offset(
+          kafka::offset_cast(*highest_translated_offset));
+        auto size_after_translated = _partition->log()->size_bytes_after_offset(
+          log_offset);
+
+        auto size_after_max_translatable
+          = _partition->log()->size_bytes_after_offset(
+            _partition->last_stable_offset());
+
+        if (size_after_translated < size_after_max_translatable) {
+            vlog(
+              datalake_log.error,
+              "Expected partition {} size after translated offset({}) {} to be "
+              "greater than or equal to the size of log after max translatable "
+              "offset({}): {}",
+              _partition->ntp().path(),
+              log_offset,
+              size_after_translated,
+              _partition->last_stable_offset());
+            return std::nullopt;
+        }
+
+        return size_after_translated - size_after_max_translatable;
+    }
+
+    std::optional<model::timestamp>
+    get_translated_offset_timestamp(kafka::offset) final {
+        return _translation_target->ts;
+    }
+
+    scheduling::clock::time_point next_checkpoint_deadline() const final {
+        auto time_left = std::chrono::milliseconds(
+                           static_cast<int>(0.9 * target_lag().count()))
```

Comment:
nitpick: this `0.9 * target_lag` snippet appears in two places. would be nice to at least bind the 0.9 portion to an identifier.